<launch>
    <arg name="camera_namespace" default="camera_center"/>

    <node name="lane_detector" pkg="nodelet" type="nodelet"
          output="screen" args="standalone lane_detector/laplacian_line_detector ">

        <!-- Thresholding -->
        <!--Range of the Laplacian Area to Floodfill-->
        <param name="laplacian_threshold_min" type="int" value="10" />
        <param name="laplacian_threshold_max" type="int" value="280" />
        <!--Strength of Adaptive Thresholding-->
        <param name="adaptive_mean_threshold" type="int" value="0" />
        <!--Minimum Area to Keep-->
        <param name="min_blob_area" type="int" value="20" />
        <param name="ignore_adaptive" type="bool" value="False" />

        <!-- ROI -->
        <!--By row #, so (larger = lower) (Image Height = 1280)-->
        <param name="lapl_ROI_top"    type="int" value="500" />
        <param name="lapl_ROI_bottom" type="int" value="860" />

        <param name="adapt_ROI_top" type="int" value="540" />
        <param name="adapt_ROI_bottom" type="int" value="750" />

        <param name="blockBumper_top" type="int" value="750" />

        <!-- BLOCKING -->
        <param name="ignore_color1_H_low"  type="int" value="0" />
        <param name="ignore_color1_S_low"  type="int" value="190" />
        <param name="ignore_color1_V_low"  type="int" value="100" />
        <param name="ignore_color1_H_high" type="int" value="255" />
        <param name="ignore_color1_S_high" type="int" value="70" />
        <param name="ignore_color1_V_high" type="int" value="255" />

        <param name="ignore_color2_H_low"  type="int" value="0" />
        <param name="ignore_color2_S_low"  type="int" value="0" />
        <param name="ignore_color2_V_low"  type="int" value="0" />
        <param name="ignore_color2_H_high" type="int" value="0" />
        <param name="ignore_color2_S_high" type="int" value="0" />
        <param name="ignore_color2_V_high" type="int" value="0" />

        <param name="ignore_gray_threshold" type="int" value="20" />

        <!-- SUB/PUB -->
        <param name="subscription_node" type="string" value="/camera_center/image_color_rect"/>
        <param name="publish_detection_node" type="string" value="lines/detection_img2"/>
        <param name="publish_debug_node" type="string" value="lines/debug_img2"/>
    </node>
</launch>
